src/Fixer/PhpUnit/PhpUnitDedicateAssertFixer.php

```diff
@@ -169,6 +171,35 @@ public function configure(array $configuration): void
                 'str_contains',
             ]);
         }
+
+        if (PhpUnitTargetVersion::fulfills($this->configuration['target'], PhpUnitTargetVersion::VERSION_9_1)) {
+            $this->fixMap = array_merge($this->fixMap, [
```

Comment:
instead, keep `fixMap` static/const and control which functions shall be fixed by `this->functions`, like for other PHPUnit versions

Reply:
Thank you for the advice. I didn't really understand how the rules worked, but now I get it. I refactored it.

Reply:
as I mentioned [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7997#discussion_r1596412684) , rename of `assertNotIsReadable` -> `assertIsNotReadable` 

This rule is:
> PHPUnit assertions like `assertInternalType`, `assertFileExists`, should be used over `assertTrue`.

so instead of `->assertTrue(is_string(...))`, we use `->assertIsString(...)`

What you aiming to do is to replace `->assertNotIsReadable($a)` with `->assertIsNotReadable($a)`, which is a diffrent thing and should be handled by different rule.
So even if John Doe will NOT want to replace `->assertTrue(is_String...`, they can still run the rule to replace renamed methods.

Reply:
@keradus Thank you for the review, I've refactored this into a separate rule.